projects/angular-ngrx-material-starter/src/app/core/utils/utils.ts

```diff
@@ -0,0 +1,44 @@
+import { Subscription, SchedulerLike } from 'rxjs';
+import { NgZone } from '@angular/core';
+
+class LeaveZoneScheduler implements SchedulerLike {
```

Comment:
This looks a bit excessive ? Why not just wrap the stream with `runOutsideOfZone` inline in the effects class?

Reply:
@tomastrajan , hey thanks for the feedback. As far as know, wrapping the interval stream with `runOutsideAngular` won't be enough since the scheduler dispatches the action within the zone anyway. That's why this utils was needed to define our own SchedulerLike to pass into the interval observable, which handled scheduling the task in/out of the zone. It's discussed a bit [here](https://stackoverflow.com/questions/43121400/run-ngrx-effect-outside-of-angulars-zone-to-prevent-timeout-in-protractor), which is where this resolution was derived.

I'm not sure if there is another way around it using observables from what I saw and tried. Alternatively we could use pure js `setInterval` to dispatch the action instead. like this perhaps:
```
  hour = 0;
  changeHour = this.ngZone.runOutsideAngular(() => 
    setInterval(() => {
      const hour = new Date().getHours();
      if (hour !== this.hour) {
        this.hour = hour;
        this.ngZone.run(() => this.store.dispatch(actionSettingsChangeHour({ hour })));
      }
    }, 3_000)
  );
```

wouldn't need the utils class for this and would work just the same sticking it in the effects (even though it wouldn't 'exactly' be an effect).

I'm still new to this, so I could be missing something obvious. Anyway, lemme know what you think of the proposed solutions.

p.s. another good [discussion](https://stackoverflow.com/questions/52018881/why-would-i-use-rxjs-interval-or-timer-polling-instead-of-window-setinterval) about why using interval observable over setInterval has some advantages, all of which may not matter too much for this app besides composability and sticking to ngrx architectural designs. 🤔

Reply:
Hi @dvargas46 !

Thank you for very detailed explanation, in that case I would rather go with much simpler `setInterval` solution, less code is always better !

Reply:
Sounds good, I have updated my PR to use the simpler approach then. Thanks!